File: app/frontend/lib/model/product.dart
import 'package:flutter/foundation.dart';
import './category.dart' as cat;

class Product {
  int id;
  String name;
  int price;
  List<dynamic> categories;

  Product(
      {required this.id,
      required this.price,
      required this.name,
      required this.categories});

  factory Product.fromJson({required Map<String, dynamic> json}) {
    return Product(
        id: json['id'],
        name: json['name'],
        price: json['price'],
        categories: json['category']);
  }
}

File: product.py
import json
class Product :

    @staticmethod
    def toJson(products):
        data = []
        ids = {}
        
        for id, price ,name ,category in products:
            if id not in ids:
                ids[id] = {}
                ids[id]['category'] = [] 
            ids[id]['id'] = id
            ids[id]['price'] = price
            ids[id]['name'] = name
            ids[id]['category'].append(category)

            # data.append(ids)
        
        for d in ids.values():
            data.append(d)

        return json.dumps(data , indent =4 )
